app/code/Magento/Customer/Controller/Plugin/Account.php

```diff
@@ -3,72 +3,78 @@
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
+declare(strict_types=1);
+
 namespace Magento\Customer\Controller\Plugin;
 
+use Closure;
+use Magento\Customer\Controller\AccountInterface;
 use Magento\Customer\Model\Session;
-use Magento\Framework\App\ActionInterface;
 use Magento\Framework\App\RequestInterface;
 use Magento\Framework\App\ResponseInterface;
-use Magento\Framework\App\Action\AbstractAction;
 use Magento\Framework\Controller\ResultInterface;
 
+/**
+ * Plugin verifies permissions using Action Name against injected (`fontend/di.xml`) rules
+ */
 class Account
 {
     /**
      * @var Session
      */
-    protected $session;
+    private $session;
+
+    /**
+     * @var RequestInterface
+     */
+    private $request;
 
     /**
      * @var array
      */
     private $allowedActions = [];
 
     /**
+     * @param RequestInterface $request
      * @param Session $customerSession
      * @param array $allowedActions List of actions that are allowed for not authorized users
      */
     public function __construct(
+        RequestInterface $request,
         Session $customerSession,
         array $allowedActions = []
     ) {
+        $this->request = $request;
```

Comment:
Even though this is a plugin and the changes are made in the scope of a major release, I think it still worth adding this dependency in [a backward-compatible way](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/#adding-a-constructor-parameter).
@lbajsarowicz what do you think?

Reply:
The plugin is not something that is expected to be inherited.
If there's a need to inherit plugin, this means that it was badly designed.

Reply:
@dmytro-ch Do you want me to introduce some change to this PR? (`needs update` status)

Reply:
Agree. Thank you!